### PR TITLE
MM-35673 Detox/E2E: Add e2e for messaging

### PR DESCRIPTION
**Pull request by josephbaylon**
#### Summary
- Added tests for MM-T100, MM-T101, MM-T107, MM-T108, MM-T115, MM-T116, MM-T128, MM-T151 in
    - `detox/e2e/test/messaging/message_deletion.e2e.js`
    - `detox/e2e/test/messaging/message_draft.e2e.js`
    - `detox/e2e/test/messaging/message_edit.e2e.js`
    - `detox/e2e/test/messaging/message_posting.e2e.js`

#### Ticket Link
JIRA tickets:
- https://mattermost.atlassian.net/browse/MM-35673

Zephyr cases:
- [MM-T100](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T100)
- [MM-T101](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T101)
- [MM-T107](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T107)
- [MM-T108](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T108)
- [MM-T115](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T115)
- [MM-T116](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T116)
- [MM-T128](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T128)
- [MM-T151](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T151)

#### Screenshots
Screen Shot 2021-05-24 at 8 33 24 AM
Screen Shot 2021-05-24 at 8 40 01 AM
Screen Shot 2021-05-24 at 8 33 16 AM
Screen Shot 2021-05-24 at 9 56 27 AM

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: detox/e2e/test/messaging/message_draft.e2e.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        // # Type a message draft
+        const message = Date.now().toString();
+        await postInput.typeText(message);
+
+        // * Verify message draft
+        await expect(postInput).toHaveText(message);
+
+        // # Send app to home and re-open
+        await device.sendToHome();
+        await device.launchApp({newInstance: false});
```
Comment (shazm) [inline detox/e2e/test/messaging/message_draft.e2e.js:66]: Should the draft still be available if the app is removed from memory (not in hibernation)?

**Review by shazm — APPROVED** [review]
Just had one question but all tests pass and app runs without issues 👍

File: detox/e2e/test/messaging/message_draft.e2e.js (hunk 1 above)
Comment (josephbaylon) [reply detox/e2e/test/messaging/message_draft.e2e.js:66]: Hi @shazm did you mean when you delete the app from device? App does not save the draft to the server so deleting the app would not save the draft.

Comment (josephbaylon) [reply detox/e2e/test/messaging/message_draft.e2e.js:66]: But if you meant closing the app (as opposed to just backgrounding), then draft should still be available.

**Review by saturninoabril — APPROVED** [review]
Thanks, LGTM. Tested and passed.

**Pull request merged**